be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1989,8 +1989,11 @@ Status SegmentIterator::_next_batch_internal(vectorized::Block* block) {
     if (UNLIKELY(!_lazy_inited)) {
         RETURN_IF_ERROR(_lazy_init());
         _lazy_inited = true;
+        uint32_t nrows_reserve_limit = _row_bitmap.cardinality() < _opts.block_row_max
+                                               ? _row_bitmap.cardinality()
```

Comment:
add comment?
std::min(_row_bitmap.cardinality(), _opts.block_row_max) is better